test/suites/container_devices_disk.sh

```diff
@@ -86,6 +86,18 @@ test_container_devices_disk_shift() {
   [ "$(lxc exec foo-isol1 -- stat /mnt/a -c '%u:%g')" = "123:456" ] || false
   [ "$(lxc exec foo-isol2 -- stat /mnt/a -c '%u:%g')" = "123:456" ] || false
 
+  mkdir -p "${TEST_DIR}/empty-dir"
+  lxc exec foo -- mkdir /opt
+  lxc config device add foo empty-dir disk source="${TEST_DIR}/empty-dir" path=/opt
+  lxc config device remove foo empty-dir
+  lxc exec foo -- stat /opt
+
+  lxc exec foo -- touch /tmp/empty-dir
+  lxc config device add foo empty-dir disk source="${TEST_DIR}/empty-dir" path=/tmp/empty-dir
+  lxc config device remove foo empty-dir
+  lxc exec foo -- stat /tmp/empty-dir
+  rm -rf "${TEST_DIR}/empty-dir"
+
```

Comment:
Please also add a third test here checking if a non existing directory gets removed after the disk device gets removed from the container (`.dir_created_by_lxd=true`). Then every case should be covered.

Reply:
Do you mean a non-existing directory on the host side or on the target-side (inside the container) ?

Reply:
Inside of the container. I see the first test checks that an existing dir in the container doesn't get removed. The second test checks if mounting over an existing file with the same name doesn't remove the file. And a third test could check if a directory gets removed if it was created by LXD explicitly for the mount.